api/client_test.go

```diff
@@ -113,3 +116,70 @@ func TestClientURL(t *testing.T) {
 		}
 	}
 }
+
+func TestDoGetFallback(t *testing.T) {
+	v := url.Values{"a": []string{"1", "2"}}
+
+	type testResponse struct {
+		Values string
+		Method string
+	}
+
+	acceptPost := true
+	// Start a local HTTP server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		req.ParseForm()
+		r := &testResponse{
+			Values: req.Form.Encode(),
+			Method: req.Method,
+		}
+
+		body, _ := json.Marshal(r)
+
+		if req.Method == http.MethodPost {
+			if !acceptPost {
```

Comment:
hm , that global var is a bit confusing, how about using the url for this.Something like:

`if req.URL.Path=="../allowPost"...}`